getBalances.js

```diff
@@ -17,28 +16,55 @@ const web3Home = new Web3(homeProvider);
 const foreignProvider = new Web3.providers.HttpProvider(FOREIGN_RPC_URL);
 const web3Foreign = new Web3(foreignProvider);
 
+const ERC20_ABI = require('./abis/ERC20.abi');
 const ERC677_ABI = require('./abis/ERC677.abi');
+const HOME_ERC_ABI = require('./abis/HomeBridgeErcToErc.abi')
 
-async function main(){
+async function main(isErcToErcMode){
   try {
-    logger.debug('calling web3Home.eth.getBalance');
-    const homeBalance = await web3Home.eth.getBalance(HOME_BRIDGE_ADDRESS)
-    const tokenContract = new web3Foreign.eth.Contract(ERC677_ABI, POA20_ADDRESS);
-    logger.debug('calling tokenContract.methods.totalSupply()');
-    const totalSupply = await tokenContract.methods.totalSupply().call()
-    const homeBalanceBN = new BN(homeBalance)
-    const foreignTotalSupplyBN = new BN(totalSupply)
-    const diff = homeBalanceBN.minus(foreignTotalSupplyBN).toString(10)
-    logger.debug("Done");
-    return {
-      home: {
-        balance: Web3Utils.fromWei(homeBalance)
-      },
-      foreign: {
-        totalSupply: Web3Utils.fromWei(totalSupply),
-      },
-      balanceDiff: Number(Web3Utils.fromWei(diff)),
-      lastChecked: Math.floor(Date.now() / 1000)
+    if (isErcToErcMode) {
+      logger.debug('calling web3Foreign.eth.getBalance');
+      const foreignBalance = await web3Foreign.eth.getBalance(FOREIGN_BRIDGE_ADDRESS)
+      const homeBridge = new web3Home.eth.Contract(HOME_ERC_ABI, HOME_BRIDGE_ADDRESS);
+      logger.debug('calling homeBridge.methods.erc677token');
+      const tokenAddress = await homeBridge.methods.erc677token().call()
+      const tokenContract = new web3Home.eth.Contract(ERC677_ABI, tokenAddress);
+      logger.debug('calling tokenContract.methods.totalSupply()');
+      const totalSupply = await tokenContract.methods.totalSupply().call()
+      const foreignBalanceBN = new BN(foreignBalance)
```

Comment:
Yes, you are right (to the note at the end of https://github.com/poanetwork/bridge-monitor/pull/9#issuecomment-418765582). For `erc20-to-erc20` mode we need to monitor balance of ERC20 for the foreign bridge contract.

Reply:
@akolotov Updated to display balance of ERC20 for the foreign bridge contract.
Example of output for erc20-erc20 mode:

```
{
    "home": {
        "totalSupply": "0.01",
        "deposits": 1,
        "withdrawals": 2,
        "requiredSignatures": 1
    },
    "foreign": {
        "erc20Balance": "0.02",
        "deposits": 1,
        "withdrawals": 3,
        "requiredSignatures": 1
    },
    "balanceDiff": 0.01,
    "lastChecked": 1536325837,
    "depositsDiff": 0,
    "withdrawalDiff": -1,
    "requiredSignaturesMatch": true
}
```